server/accounts.go

```diff
@@ -3634,29 +3636,48 @@ func (s *Server) updateAccountClaimsWithRefresh(a *Account, ac *jwt.AccountClaim
 			if acc.Name == a.Name {
 				return true
 			}
-			// TODO: checkServiceImportAuthorized() stack should not be trying
-			// to lock "acc". If we find that to be needed, we will need to
-			// rework this to ensure we don't lock acc.
-			acc.mu.Lock()
+			type siState struct {
+				invalid  bool
+				response bool
+				latency  *serviceLatency
+				atrc     bool
+			}
+			sivs := map[*serviceImport]siState{}
+			acc.mu.RLock()
 			for _, sis := range acc.imports.services {
 				for _, si := range sis {
 					if si != nil && si.acc.Name == a.Name {
-						// Check for if we are still authorized for an import.
-						si.invalid = !a.checkServiceImportAuthorized(acc, si.to, si.claim)
+						state := siState{
+							invalid:  !a.checkServiceImportAuthorized(acc, si.to, si.claim),
+							response: si.response,
+						}
 						// Make sure we should still be tracking latency and if we
 						// are allowed to trace.
-						if !si.response {
-							if se := a.getServiceExport(si.to); se != nil {
+						if !state.response {
+							a.mu.RLock()
+							se := a.getServiceExport(si.to)
```

Comment:
Should we have this instead, so that access to `se.` fields is still protected?
```
a.mu.RLock()
if se:=a.getServiceExport(si.to); se != nil {
    if is.latency != nil {
        state.latency = se.latency
    }
    // Update allow trace.
    state.atrc = se.atrc
}
a.mu.RUnlock()
```

Reply:
Good spot, have extended the lock around that.